authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,1006 @@
+  }
+
+  /** Create a Ranger role if the role does not exist. */
+  private RangerRole createRangerRoleIfNotExists(String roleName) {
+    RangerRole rangerRole = null;
+    try {
+      rangerRole = rangerClient.getRole(roleName, RANGER_ADMIN_NAME, rangerServiceName);
+    } catch (RangerServiceException e) {
+      // ignore exception, If the role does not exist, then create it.
+    }
+    try {
+      if (rangerRole == null) {
+        rangerRole = new RangerRole(roleName, MANAGED_BY_GRAVITINO, null, null, null);
+        rangerClient.createRole(rangerServiceName, rangerRole);
+      }
+    } catch (RangerServiceException e) {
+      throw new RuntimeException(e);
+    }
+    return rangerRole;
+  }
+
+  /**
+   * Find the managed policy for the metadata object.
+   *
+   * @param metadataObject The metadata object to find the managed policy.
+   * @return The managed policy for the metadata object.
+   */
+  @VisibleForTesting
+  public RangerPolicy findManagedPolicy(MetadataObject metadataObject)
+      throws AuthorizationPluginException {
+    List<String> nsMetadataObj =
+        Lists.newArrayList(SecurableObjects.DOT_SPLITTER.splitToList(metadataObject.fullName()));
+    nsMetadataObj.remove(0); // skip `catalog`
+    Map<String, String> policyFilter = new HashMap<>();
+    Map<String, String> preciseFilterKeysFilter = new HashMap<>();
+    policyFilter.put(RangerDefines.SEARCH_FILTER_SERVICE_NAME, this.rangerServiceName);
+    policyFilter.put(SearchFilter.POLICY_LABELS_PARTIAL, MANAGED_BY_GRAVITINO);
+    for (int i = 0; i < nsMetadataObj.size(); i++) {
+      policyFilter.put(policyFilterKeys.get(i), nsMetadataObj.get(i));
+      preciseFilterKeysFilter.put(policyPreciseFilterKeys.get(i), nsMetadataObj.get(i));
+    }
+
+    try {
+      List<RangerPolicy> policies = rangerClient.findPolicies(policyFilter);
+
+      if (!policies.isEmpty()) {
+        // Because Ranger doesn't support the precise filter, Ranger will return the policy meets
+        // the wildcard(*,?) conditions, just like `*.*.*` policy will match `db1.table1.column1`
+        // So we need to manual precise filter the policies.
+        policies =
+            policies.stream()
+                .filter(
+                    policy ->
+                        policy.getResources().entrySet().stream()
+                            .allMatch(
+                                entry ->
+                                    preciseFilterKeysFilter.containsKey(entry.getKey())
+                                        && entry.getValue().getValues().size() == 1
+                                        && entry
+                                            .getValue()
+                                            .getValues()
+                                            .contains(preciseFilterKeysFilter.get(entry.getKey()))))
+                .collect(Collectors.toList());
+      }
+
+      // Only return the policies that are delegate Gravitino management
```

Comment:
Only return the policies that are delegate Gravitino management
```suggestion
      // Only return the policies that are managed by Gravitino.
```
Comment